frontend/src/components/dashboard/aliases/BlockLevelSlider.tsx

```diff
@@ -205,6 +242,169 @@ const BlockLevelIllustration = (props: { level: BlockLevel }) => {
+    >
+      {props.children}
+    </div>
+  );
+};
+
+type PromotionalTooltipProps = {
+  onClose: () => void;
+  triggerRef: React.RefObject<HTMLButtonElement>;
+  overlayProps: HTMLAttributes<HTMLDivElement>;
+  premiumAvailableInCountry: boolean;
+};
+const PromotionalTooltip = (props: PromotionalTooltipProps) => {
+  const { l10n } = useLocalization();
+  const overlayRef = useRef<HTMLDivElement>(null);
+  const { overlayProps, underlayProps } = useOverlay(
+    { isOpen: true, onClose: props.onClose, isDismissable: true },
+    overlayRef
+  );
+
+  const { modalProps } = useModal();
+
+  const { dialogProps, titleProps } = useDialog({}, overlayRef);
+
+  const overlayPositionProps = useOverlayPosition({
+    targetRef: props.triggerRef,
+    overlayRef: overlayRef,
+    placement: "bottom left",
+    offset: 60,
+  }).overlayProps;
+
+  const closeButtonRef = useRef<HTMLButtonElement>(null);
+  const closeButtonProps = useButton(
+    {
+      onPress: () => props.onClose(),
+    },
+    closeButtonRef
+  ).buttonProps;
+
+  const link = props.premiumAvailableInCountry ? (
+    <Link href="/premium/">
+      <a>
+        {l10n.getString(
+          "profile-promo-email-blocking-description-promotionals-locked-cta"
+        )}
+      </a>
+    </Link>
+  ) : (
+    <Link href="/premium/waitlist">
+      <a>
+        {l10n.getString(
+          "profile-promo-email-blocking-description-promotionals-locked-waitlist-cta"
+        )}
+      </a>
+    </Link>
+  );
+
+  return (
+    <div {...underlayProps} className={styles["upgrade-tooltip-underlay"]}>
+      <FocusScope restoreFocus contain autoFocus>
+        <div
+          {...mergeProps(
+            overlayProps,
+            dialogProps,
+            props.overlayProps,
+            overlayPositionProps,
+            modalProps
+          )}
+          ref={overlayRef}
+          className={styles["upgrade-tooltip"]}
+        >
+          <img src={UmbrellaSemi.src} alt="" />
+          <span className={styles["upgrade-message"]}>
+            <b className={styles["locked-message"]} {...titleProps}>
+              <LockIcon alt="" className={styles["lock-icon"]} />
+              {l10n.getString(
+                "profile-promo-email-blocking-description-promotionals-locked-label"
+              )}
+            </b>
+            {l10n.getString(
+              "profile-promo-email-blocking-description-promotionals"
+            )}
+            {link}
+          </span>
+          <button
+            {...closeButtonProps}
+            ref={closeButtonRef}
+            className={styles["close-button"]}
+          >
+            <CloseIcon
+              alt={l10n.getString(
+                "profile-promo-email-blocking-description-promotionals-locked-close"
+              )}
+            />
+          </button>
+        </div>
+      </FocusScope>
+    </div>
+  );
+};
+
 function getSliderValueForAlias(alias: AliasData): number {
   if (alias.enabled === false) {
     return 100;
```

Comment:
Praise: I'm not sure if I mentioned this in a previous review, but this slider really is slick.

Reply:
Thanks in large part to react-aria! (Although there's a slight impedance mismatch, given that from the user's point of view they're not _really_ setting a number value.)